src/factories/ClaimConditionFactory.ts

```diff
@@ -141,16 +148,23 @@ class ClaimConditionFactory {
    *
    * @param phase - The phase to remove
    */
-  public removeClaimPhase(index: number): void {
+  public async deleteClaimPhase(index: number): Promise<void> {
     if (index < 0 || index >= this.phases.length) {
       return;
     }
 
-    const sorted = this.buildConditions();
+    const sorted = await this.buildConditions();
     const cleared = sorted.splice(index - 1, 1);
     this.fromPublicClaimConditions(cleared);
   }
 
+  /**
+   * @deprecated - Use {@link deleteClaimPhase} instead.
+   */
+  public removeClaimPhase(_index: number): void {
+    throw new FunctionDeprecatedError("deleteClaimPhase");
```

Comment:
not too big of a fan of this. in general we need to keep deprecated function working.
one thought is that we can offload the removed index until the end, build conditions time.

Reply:
could you elaborate a bit more?